pkg/pb/synthetic_monitoring/checks_extra.go

```diff
@@ -418,6 +418,69 @@ func (c Check) ConfigVersion() string {
 	return strconv.FormatInt(int64(c.Modified*1000000000), 10)
 }
 
+var (
+	// This pattern matches "k6/secrets" or 'k6/secrets'. This is the basic canary to tell us if
+	// a script might be using secrets.
+	secretsCanary = `(?:'k6/secrets'|"k6/secrets")`
+
+	usesSecretsModuleRe = regexp.MustCompile(
+		`(?m:` +
+			// Default import: import secrets from 'k6/secrets'
+			`^\s*import\s+secrets\s+from\s+` + secretsCanary +
+			`|` +
+			// Named import: import { } from 'k6/secrets'
+			// Named import: import { secrets } from 'k6/secrets'
+			// Named import: import { secrets, foo } from 'k6/secrets'
+			// Named import: import { foo, secrets } from 'k6/secrets'
+			// Named import: import { foo, secrets, bar } from 'k6/secrets'
+			// Named import: import { secrets as s } from 'k6/secrets'
+			`^\s*import\s*{[^}]*}\s+from\s+` + secretsCanary +
+			`|` +
+			// Namespace import: import * as secrets from 'k6/secrets
+			`^\s*import\s+\*\s+as\s+\S+\s+from\s+` + secretsCanary +
+			`|` +
+			// Require syntax: const secrets = require('k6/secrets')
+			`^\s*const\s+secrets\s*=\s*require\s*\(\s*` + secretsCanary + `\s*\)` +
+			`)`)
+)
```

Comment:
This is the bit that is actually interesting.

I have tried to keep the regular expression simple.

It's matching lines (the `?m:` bit at the start, so it's possible to write `^\s*import`, meaning it's OK to have blanks before `import`, but `import` must be the first thing on the line.